command/genesis/utils.go

```diff
@@ -119,8 +122,47 @@ func getValidatorsFromPrefixPath(prefix string) ([]types.Address, error) {
 			return nil, err
 		}
 
-		validators = append(validators, crypto.PubKeyToAddress(&priv.PublicKey))
+		privateKeys = append(privateKeys, priv)
 	}
 
-	return validators, nil
+	switch keyType {
+	case crypto.KeySecp256k1:
+		return toECDSAValidatorSet(privateKeys), nil
+	case crypto.KeyBLS:
+		return toBLSValidatorSet(privateKeys)
+	default:
+		return nil, fmt.Errorf("invalid key type: %s", keyType)
+	}
+}
+
+func toECDSAValidatorSet(keys []*ecdsa.PrivateKey) validators.ValidatorSet {
+	addrs := make([]types.Address, len(keys))
+
+	for idx, key := range keys {
+		addrs[idx] = crypto.PubKeyToAddress(&key.PublicKey)
+	}
+
+	valSet := validators.ECDSAValidatorSet(addrs)
+
+	return &valSet
+}
+
+func toBLSValidatorSet(keys []*ecdsa.PrivateKey) (validators.ValidatorSet, error) {
+	vals := make([]validators.BLSValidator, len(keys))
+
+	for idx, key := range keys {
+		pubkey, err := crypto.ECDSAToBLSPubkey(key)
+		if err != nil {
+			return nil, err
+		}
+
+		vals[idx] = validators.BLSValidator{
+			Address:   crypto.PubKeyToAddress(&key.PublicKey),
+			BLSPubKey: pubkey,
+		}
+	}
+
+	valSet := validators.BLSValidatorSet(vals)
+
+	return &valSet, nil
```

Comment:
What do you think about moving this out to a more appropriate place?

Reply:
Good catch. I moved in 54c8009a3f1e8d324502565a9aff5bf57d3697b0